entrypoint.sh

```diff
@@ -3,98 +3,239 @@
 if [[ "$LINES_CHANGED" = "0" ]] && [[ ! "$INPUT_PR_ALLOW_EMPTY" ==  "true" ]]; then
-  echo "No file changes detected between source and destination branches."
+  echo_blue "No file changes detected between source and destination branches."
   exit 0
 fi
+echo_yellow "Source and Destination Branches Contain Differences."
+echo_yellow "Number of lines changed: $LINES_CHANGED"
+echo "::endgroup::"
 
 
+echo "::group::Establish hub cli parameters"
 # Workaround for `hub` auth error https://github.com/github/hub/issues/2149#issuecomment-513214342
 export GITHUB_USER="$GITHUB_ACTOR"
+# set GITHUB_TOKEN envar so hub cli commands can authenticate.
+export GITHUB_TOKEN="$INPUT_TOKEN"
 
-PR_ARG="$INPUT_PR_TITLE"
-if [[ ! -z "$PR_ARG" ]]; then
-  PR_ARG="-m \"$PR_ARG\""
+if [[ ! -z "$DESTINATION_BRANCH" ]]; then
+  FLAGS=(-b "$DESTINATION_BRANCH")
+fi
+
+if [[ ! -z "$SOURCE_BRANCH" ]]; then
+  FLAGS+=(-h "$SOURCE_BRANCH")
+fi
 
+FLAGS+=(--no-edit)
+
+if [[ ! -z "$INPUT_PR_TITLE" ]]; then
+  FLAGS+=(-m "$INPUT_PR_TITLE")
   if [[ ! -z "$INPUT_PR_TEMPLATE" ]]; then
     sed -i 's/`/\\`/g; s/\$/\\\$/g' "$INPUT_PR_TEMPLATE"
-    PR_ARG="$PR_ARG -m \"$(echo -e "$(cat "$INPUT_PR_TEMPLATE")")\""
+    FLAGS+=(-m "$(echo -e "$(cat "$INPUT_PR_TEMPLATE")")")
   elif [[ ! -z "$INPUT_PR_BODY" ]]; then
-    PR_ARG="$PR_ARG -m \"$INPUT_PR_BODY\""
+    FLAGS+=(-m "$INPUT_PR_BODY")
   fi
 fi
 
 if [[ ! -z "$INPUT_PR_REVIEWER" ]]; then
-  PR_ARG="$PR_ARG -r \"$INPUT_PR_REVIEWER\""
+  FLAGS+=(-r "$INPUT_PR_REVIEWER")
 fi
 
 if [[ ! -z "$INPUT_PR_ASSIGNEE" ]]; then
-  PR_ARG="$PR_ARG -a \"$INPUT_PR_ASSIGNEE\""
+  FLAGS+=(-a "$INPUT_PR_ASSIGNEE")
 fi
 
 if [[ ! -z "$INPUT_PR_LABEL" ]]; then
-  PR_ARG="$PR_ARG -l \"$INPUT_PR_LABEL\""
+  FLAGS+=(-l "$INPUT_PR_LABEL")
 fi
 
 if [[ ! -z "$INPUT_PR_MILESTONE" ]]; then
-  PR_ARG="$PR_ARG -M \"$INPUT_PR_MILESTONE\""
+  FLAGS+=(-M "$INPUT_PR_MILESTONE")
 fi
 
 if [[ "$INPUT_PR_DRAFT" ==  "true" ]]; then
-  PR_ARG="$PR_ARG -d"
+  FLAGS+=(-d)
 fi
 
-COMMAND="hub pull-request \
-  -b $DESTINATION_BRANCH \
-  -h $SOURCE_BRANCH \
-  --no-edit \
-  $PR_ARG \
-  || true"
+echo "${FLAGS[@]}"
+echo "::endgroup::"
+
 
+echo "::group::Create Pull-Request $SOURCE_BRANCH -> $DESTINATION_BRANCH"
+RAND_UUID=$(cat /proc/sys/kernel/random/uuid)
+COMMAND="hub pull-request "${FLAGS[@]}" 2>\"./create-pull-request.$RAND_UUID.stderr\" || true"
 echo "$COMMAND"
 
-PR_URL=$(sh -c "$COMMAND")
-if [[ "$?" != "0" ]]; then
-  exit 1
+PR_URL=$( \
+  hub pull-request "${FLAGS[@]}" \
+  2>"./create-pull-request.$RAND_UUID.stderr" || true \
+)
+STD_ERROR="$( cat "./create-pull-request.$RAND_UUID.stderr" || true )"
+rm -rf "./create-pull-request.$RAND_UUID.stderr"
+echo "::endgroup::"
+
+
+echo "::group::Revert Git Config Changes"
+# set origin back as was previously configured.
+git remote set-url origin "$GIT_ORIGIN_URL"
+git fetch origin '+refs/heads/*:refs/heads/*' --update-head-ok
+git fetch --prune
+echo "::endgroup::"
```

Comment:
Instead of reverting Git config, should we drop the requirement of running `actions/checkout` and just perform checkout in a non-GITHUB_WORKSPACE directory? That way we cut down on a step and we don't need to revert git config here either.

Reply:
Do you know if its possible to clone outside of GITHUB_WORKSPACE, or even write any files? So there's a few things that I don't want to have to deal with, even if it would make this action more streamlined/slick.

Reply:
Yeah you can clone either outside of it in a temp folder (which will be removed in next job step) or in a subfolder of GITHUB_WORKSPACE which will be persisted for next steps to use.

We don't have to make this modification for now as what we have currently works.

Reply:
I'd rather put the onus on the user to successfully checkout their code before needlessly involving resources here and constantly fielding questions and issues related to something that really should have nothing to do with this action. Yes, it would be nice to implement, and make the workflow easier to read, ect,ect, but I wouldn't be willing to expand the scope of work unless there is a real need.

Reply:
Thanks for the input, I feel the same way. 👍